extras/catch_amalgamated.cpp

```diff
@@ -8461,6 +8463,108 @@ namespace Catch {
 
 
 
+#include <ostream>
+
+namespace Catch {
+
+    namespace {
+        std::ostream& tagWriter( std::ostream& out,
+                                 std::vector<Tag> const& tags ) {
+            out << ';';
+            if ( tags.empty() ) { return out; }
+
+            out << "[=[" << "[==[" << tags.front().original << "]==]";
```

Comment:
Can the number of equal signs be a variable, just in case? The brackets should also be limited to where we want to escape the strings, i.e. name, tag, etc. If we go with writing a full script it would be even more minimal.

An example script could be
```
set(test_data)
set(test_name [[<insert_test_name>]])
set(test_file [[<insert_test_file>]])
set(test_tags [[<insert_test_tag_0>]] ...)
list(APPEND test_data "${test_name};${test_file};${test_tags}")
```

You might need to play around with `join`/`split` and the need for escaping or not for list parameters like `test_tags`